core/src/main/java/org/apache/accumulo/core/spi/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -518,6 +518,8 @@ public long balance(BalanceParameters params) {
 
       migrationsOut.addAll(newMigrations);
       if (migrationsOut.size() >= myConf.maxTServerMigrations) {
+        LOG.warn("Table {} migration size : {} is over tserver migration max: {}", tableId,
```

Comment:
I'm wondering if this should be `debug` instead of `warn`. I think the code will process this batch of migrations because it's exceeded the batch size, but I'm pretty sure that it will come right back to this code and get the migrations for the rest of the tables.

Reply:
Could this end up being a spammy message?

Reply:
> I'm wondering if this should be `debug` instead of `warn`. I think the code will process this batch of migrations because it's exceeded the batch size, but I'm pretty sure that it will come right back to this code and get the migrations for the rest of the tables.
We could drop it to debug as the information is still valuable when troubleshooting migration delay issues. 


> Could this end up being a spammy message?
At warn it could be spammy in the monitor as it would pop out on every migration run. 
The current logs are extremely spammy if migrations are happening so I don't think it would be noticed at the debug level.

Reply:
added caching for the warn level message and dropped this down to debug by default.